src/ReactiveDomain/ReactiveDomain.Tests/Logging/when_logging_high_volume_message_traffic.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using ReactiveDomain.Bus;
+using ReactiveDomain.EventStore;
+using ReactiveDomain.Messaging;
+using ReactiveDomain.Tests.Helpers;
+using ReactiveDomain.Tests.Subscribers.QueuedSubscriber;
+using Xunit;
+
+namespace ReactiveDomain.Tests.Logging
+{
+    // ReSharper disable once InconsistentNaming
+    public class when_logging_high_volume_message_traffic :
+        with_message_logging_enabled,
+        IHandle<Message>
+    {
+        private readonly Guid _correlationId = Guid.NewGuid();
+        private IListener _listener;
+
+        private readonly int _maxCountedMessages = 10000;
+
+        private int _commandFireCount;
+        private int _commandAckCount;
+        private int _commandSuccessCount;
+        private int _lastCommandCount;
+        private int _countedEventCount;
+        private int _testDomainEventCount;
+        private int _numberOfItemsLogged;
+        private int _catchupSubscriptionMsgs;
+
+        private TestCommandSubscriber _cmdHandler;
+        protected override void When()
+        {
+            // commands must have a commandHandler
+            _cmdHandler = new TestCommandSubscriber(Bus);
+
+             _commandFireCount = 0;
+             _commandAckCount = 0;
+             _commandSuccessCount = 0;
+             _lastCommandCount = 0;
+             _countedEventCount = 0;
+             _testDomainEventCount = 0;
+             _numberOfItemsLogged = 0;
+             _catchupSubscriptionMsgs = 0;
+
+            _listener = Repo.GetListener(Logging.FullStreamName);
+            _listener.EventStream.Subscribe<Message>(this);
+
+            _listener.Start(Logging.FullStreamName);
+
+            // create and fire a mixed set of commands and events
+            for (int i = 0; i < _maxCountedMessages; i++)
+            {
+                Bus.Publish(
+                    new CountedEvent(i,
+                        Guid.NewGuid(),
+                        Guid.NewGuid()));
+
+                // this is just an example command - choice to fire this one was random
+                var cmd = new InformUserCmd("title",
+                                        $"message{i}",
+                                        Guid.NewGuid(),
+                                        null);
+                Bus.Fire(cmd,
+                    $"exception message{i}",
+                    TimeSpan.FromSeconds(2));
+
+                Bus.Publish(new TestDomainEvent(Guid.NewGuid(), Guid.NewGuid()));
+
+            }
+
+            var tstCmd = new TestCommands.TestCommand3(
+                        _correlationId,
+                        null);
+
+            Bus.Fire(tstCmd,
+                "TestCommand3 failed",
+                TimeSpan.FromSeconds(2));
+
+        }
+
+
+        [Fact(Skip = "pending deletion of log stream")]
+        public void can_verify_high_volume_traffic_logged()
```

Comment:
"all_messages_are_logged" perhaps? The point of this one, after all, is to make sure that all messages are logged, even in the case of a high volume of message traffic.